e2e_tests/tests/cluster/test_agent_restart.py

```diff
@@ -224,3 +262,149 @@ def test_agent_restart_cmd_container_failure(
         raise
     else:
         managed_cluster.restart_agent()
+
+
+@pytest.mark.managed_devcluster
+@pytest.mark.parametrize("command_duration", [20])
+def test_agent_noreattach_restart_kills(
+    managed_cluster: ManagedCluster, command_duration: int
+) -> None:
+    if managed_cluster.reattach:
+        pytest.skip()
+
+    managed_cluster.ensure_agent_ok()
+    try:
+        command_id = run_command(command_duration)
+        wait_for_command_state(command_id, "RUNNING", 10)
+
+        for _i in range(10):
+            container_ids = list(_local_container_ids_for_command(command_id))
+            if len(container_ids) > 0:
+                break
+            time.sleep(1)
+        else:
+            pytest.fail(f"Failed to find the command container id after {_i} ticks")
+
+        managed_cluster.kill_agent()
+        managed_cluster.restart_agent(wait_for_amnesia=False)
+
+        # That command container should be killed right away.
+        for _i in range(3):
+            container_ids = list(_local_container_ids_for_command(command_id))
+            if len(container_ids) == 0:
+                break
+            time.sleep(1)
+        else:
+            pytest.fail(f"Command container wasn't killed after {_i} ticks")
+
+        assert "success" not in get_command_info(command_id)["exitStatus"]
+    except Exception:
+        managed_cluster.restart_agent()
+        raise
+
+
+@pytest.mark.managed_devcluster
+@pytest.mark.parametrize("slots", [0, 1])
+@pytest.mark.parametrize("downtime", [0, 20, 60])
+def test_agent_restart_recover_cmd(
+    managed_cluster: ManagedCluster, slots: int, downtime: int
+) -> None:
+    if not managed_cluster.reattach:
+        pytest.skip()
+
+    managed_cluster.ensure_agent_ok()
+    try:
+        command_id = run_command(30, slots=slots)
+        wait_for_command_state(command_id, "RUNNING", 10)
+
+        managed_cluster.kill_agent()
+        time.sleep(downtime)
+        managed_cluster.restart_agent(wait_for_amnesia=False)
+
+        wait_for_command_state(command_id, "TERMINATED", 30)
+
+        # Commands fail if they have finished while the agent was off.
+        reattach_wait = managed_cluster.fetch_config_reattach_wait()
+        succeeded = "success" in get_command_info(command_id)["exitStatus"]
+        assert succeeded is (reattach_wait > downtime)
```

Comment:
why is?
```suggestion
        assert succeeded == (reattach_wait > downtime)
```

Reply:
mypy tends to recommend using `is` for boolean / None checks.

Reply:
had no idea, gtk